MySQLdb/times.py

```diff
@@ -48,25 +48,33 @@ def format_TIMESTAMP(d):
 
 
 def DateTime_or_None(s):
-    if ' ' in s:
-        sep = ' '
-    elif 'T' in s:
-        sep = 'T'
-    else:
-        return Date_or_None(s)
-
     try:
-        d, t = s.split(sep, 1)
-        if '.' in t:
-            t, ms = t.split('.',1)
-            ms = ms.ljust(6, '0')
+        if len(s) < 11:
+            return Date_or_None(s)
         else:
-            ms = 0
-        return datetime(*[ int(x) for x in d.split('-')+t.split(':')+[ms] ])
-    except (SystemExit, KeyboardInterrupt):
-        raise  # pragma: no cover
-    except:
-        return Date_or_None(s)
+            micros = s[20:]
+
+            if len(micros) == 0:
+                # 12:00:00
+                micros = 0
+            elif len(micros) < 7:
+                # 12:00:00.123456
+                micros = int(micros) * 10 ** (6 - len(micros))
+            else:
+                # 12:00:00.123456789
+                micros = int(micros)
```

Comment:
> We could ofc change this behaviour since MySQL will never return more than 6 anyway. If someone is using this function for higher precision it would probably be seen as a bug returning None. What do you think?

I don't know how these module are used in the world.
Since mysqlclient is "drop-in replacement" for MySQL-python, I’m timid to change behavior.